features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/utils/PathUtils.kt

```diff
@@ -0,0 +1,178 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.utils
+
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import android.graphics.Matrix
+import android.graphics.Paint
+import android.graphics.Path
+import android.graphics.PathMeasure
+import android.graphics.RectF
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.sessionreplay.internal.recorder.resources.BitmapCachesManager
+import com.datadog.android.sessionreplay.internal.recorder.resources.HashGenerator
+import com.datadog.android.sessionreplay.internal.recorder.resources.MD5HashGenerator
+import com.datadog.android.sessionreplay.recorder.wrappers.BitmapWrapper
+import com.datadog.android.sessionreplay.recorder.wrappers.CanvasWrapper
+
+internal class PathUtils(
+    private val logger: InternalLogger = InternalLogger.UNBOUND,
+    private val bitmapCachesManager: BitmapCachesManager,
+    private val canvasWrapper: CanvasWrapper = CanvasWrapper(logger),
+    private val bitmapWrapper: BitmapWrapper = BitmapWrapper(),
+    private val md5Generator: HashGenerator = MD5HashGenerator(logger)
+) {
+    internal fun convertPathToBitmap(
+        checkPath: Path,
+        checkmarkColor: Int,
+        desiredWidth: Int,
+        desiredHeight: Int,
+        strokeWidth: Int
+    ): Bitmap? {
+        val scaledPath = scalePathToTargetDimensions(checkPath, desiredWidth, desiredHeight)
+
+        val mutableBitmap = bitmapCachesManager.getBitmapByProperties(
+            width = desiredWidth,
+            height = desiredHeight,
+            config = Bitmap.Config.ARGB_8888
+        )
+            ?: bitmapWrapper.createBitmap(desiredWidth, desiredHeight, Bitmap.Config.ARGB_8888)
+            ?: return null
+
+        return drawPathOntoBitmap(
+            bitmap = mutableBitmap,
+            scaledPath = scaledPath,
+            strokeWidth = strokeWidth,
+            checkmarkColor = checkmarkColor
+        )
+    }
+
+    private fun drawPathToBitmap(
+        checkmarkColor: Int,
+        path: Path,
+        targetStrokeWidth: Int,
+        canvas: Canvas?
+    ) {
+        val paint = Paint().apply {
+            color = checkmarkColor
+            style = Paint.Style.STROKE
+            strokeWidth = targetStrokeWidth.toFloat()
+            isAntiAlias = true
+        }
+
+        // Draw the Path onto the Canvas
+        drawPathSafe(canvas, path, paint)
+    }
+
+    private fun scalePathToTargetDimensions(
+        path: Path,
+        targetWidth: Int,
+        targetHeight: Int
+    ): Path {
+        // path initial bounds
+        val originalBounds = RectF()
+
+        @Suppress("DEPRECATION") // the new api is flagged as unstable
```

Comment:
it would be nice to have a JIRA ticket to use new API when it is not flagged anymore, otherwise we may forget about this one